Fix total count for record list

vicpatel opened the pull request:
Please take a look.

<!--
Checklist:

1. API changes have been discussed,
2. Source code must be formatted (`go fmt`),
3. Codegen shouldn't produce modified files,
4. Builds pass,
5. Tests pass,
6. Linked to relevant issues

When you are writing pull request title and describing changes, follow
commit message format in the CONTRIBUTING.md in the codebase root.

-->

darh commented on compose/dalutils/records.go:
```diff
@@ -198,6 +201,72 @@ func drainIterator(ctx context.Context, iter dal.Iterator, mod *types.Module, f
+				}
+			}
+
+			// total = 0
```
we probably don't need this commented-out section?

vicpatel committed:
Fix total and page navigation for records

Introduce helper func(IteratorPaging) to count total and create page navigation for records and also fixes timezone and milliseconds fir dateTime field value for query.

tjerman reviewed (commented):
lgtm; there are some commented out variables in case you forgot about those

The other ticket (not seeing the full paging cursor list) will also need work on similar places, so worth doing both in one go?
